llama-index-integrations/storage/chat_store/llama-index-storage-chat-store-redis/llama_index/storage/chat_store/redis/base.py

```diff
@@ -20,25 +27,31 @@ def _message_to_dict(message: ChatMessage) -> dict:
 
 # Convert the json object in Redis to a ChatMessage
 def _dict_to_message(d: dict) -> ChatMessage:
-    return ChatMessage.parse_obj(d)
+    return ChatMessage.model_validate(d)
 
 
 class RedisChatStore(BaseChatStore):
     """Redis chat store."""
 
     redis_client: Any = Field(description="Redis client.")
+    aredis_client: Any = Field(default=None, description="Async Redis client.")
     ttl: Optional[int] = Field(default=None, description="Time to live in seconds.")
 
     def __init__(
         self,
         redis_url: str = "redis://localhost:6379",
-        redis_client: Optional[Any] = None,
+        redis_client: Optional[Redis] = None,
+        aredis_client: Optional[AsyncRedis] = None,
         ttl: Optional[int] = None,
         **kwargs: Any,
     ) -> None:
         """Initialize."""
         redis_client = redis_client or self._get_client(redis_url, **kwargs)
```

Comment:
does this work? For some reason, I always thought using `self.` before calling super on pydantic objects would cause pydantic errors?

Reply:
I didn't come across that issue while testing, but I changed to call `super` before using `self.`